RELEASES.md

```diff
@@ -68,7 +68,19 @@ Compatibility Notes
 - [Rustc now correctly relates the lifetime of an existential associated
   type.][71896] This fixes some edge cases where `rustc` would erroneously allow
   you to pass a shorter lifetime than expected.
-
+- [Rustc now dynamically links to `libz` (also called `zlib`).][74420] The
+  library will need to be installed for `rustc` to work, even though we expect
+  it to be already available on most systems.
```

Comment:
Note that we don't link on all platforms, IIRC just linux? But it seems fine to over-warn...

Reply:
It's also on windows-gnu but statically so there is no additional dependency.